### DISCUSSION TITLE
why am_zcopy has iov cnt = 3 limit?

### DISCUSSION THREAD

**Discussion by haipeng31** [openucx/ucx#7247] [Q&A] [answered]
#             am_zcopy: <= 8254, up to 3 iov

it seems that hardware has up to 30 iov cnt limit
ibv_devinfo -v | grep sge
	max_sge:			30

**Comment by yosefe** [comment, marked as answer]
UCT layer shares the WQE space between IOV entries and AM header. currently the number of IOV is hard-coded to 3 and the rest is for am_zcopy header to be sent as inline data.

**Reply by haipeng31** [reply]
many thanks, yosefe.
